sympy/physics/mechanics/joint.py

```diff
@@ -1046,3 +1079,302 @@ def _set_linear_velocity(self):
         self.child_point.set_vel(self.child.frame, 0)
         self.child_point.set_vel(self.parent.frame, self.speeds[0] * axis)
         self.child.masscenter.set_vel(self.parent.frame, self.speeds[0] * axis)
+
+
+class CylindricalJoint(_JointAxisMixin, Joint):
+    """Cylindrical Joint.
+
+    .. image:: CylindricalJoint.svg
+        :align: center
+        :width: 600
+
+    Explanation
+    ===========
+
+    A cylindrical joint is defined such that the child body rotates about as
+    well as translates along the body fixed joint axis with respect to the
+    parent body. So this joint axis is the the rotation axis as well as the
+    translation axis. The location of the joint is defined by two points, one in
+    each body, which coincide when the generalized coordinate corresponding to
+    the translation is zero. The direction cosine matrix between the
+    childinterframe and parentinterframe is formed using a simple rotation about
+    the joint axis. The page on the joints framework gives a more detailed
+    explanation on the intermediate frames.
+
+    Parameters
+    ==========
+
+    name : string
+        A unique name for the joint.
+    parent : Body
+        The parent body of joint.
+    child : Body
+        The child body of joint.
+    rotation_coordinate : dynamicsymbol, optional
+        Generalized coordinate corresponding to the rotation angle. The default
+        value is ``dynamicsymbols(f'q0_{joint.name}')``.
+    translation_coordinate : dynamicsymbol, optional
+        Generalized coordinate corresponding to the translation distance. The
```

Comment:
"displacement" over "distance" as this is a vector quantity?
```suggestion
        Generalized coordinate corresponding to the translation displacement. The
```

Reply:
Another slight pro of distance over displacement is that it is a bit shorter, so that the description of the `translation_coordinate` property fits within the 80 characters.